components/extensions/org.wso2.carbon.ml.siddhi.extension/src/main/java/org/wso2/carbon/ml/siddhi/extension/streamingml/samoa/StreamingClusteringTaskBuilder.java

```diff
@@ -0,0 +1,173 @@
+                Integer.MAX_VALUE);
+
+        Option[] extraOptions = new Option[] { suppressStatusOutOpt, suppressResultOutOpt, statusUpdateFreqOpt };
+
+        StringBuilder cliString = new StringBuilder();
+        for (String arg : args) {
+            logger.info(arg);
+            cliString.append(" ").append(arg);
+        }
+        logger.debug("Command line string = {}"+cliString.toString());
+        System.out.println("Command line string = " + cliString.toString());
+
+
+        Task task;
+        try {
+            task = ClassOption.cliStringToObject(cliString.toString(), Task.class, extraOptions);
+            logger.info("Successfully instantiating {}"+task.getClass().getCanonicalName());
+        } catch (Exception e) {
+            logger.error("Fail to initialize the task", e);
+            logger.info("Fail to initialize the task" + e);
+            return;
+        }
+
+        task.setFactory(new SimpleComponentFactory());
+        task.init();
+        SimpleEngine.submitTopology(task.getTopology());
+    }
+
+
+    public void initTask(int numAttributes, int numClusters, int batchSize, int maxNumEvents){
+        String query="";
+        //query ="org.gsoc.samoa.streaming.samoa.StreamingClusteringTask -f "+batchSize+" -i "+maxNumEvents+" -s  (org.gsoc.samoa.streaming.samoa.StreamingClusteringStream -K "+numClusters+" -a "+numAttributes+") -l (org.apache.samoa.learners.clusterers.simple.DistributedClusterer -l (org.apache.samoa.learners.clusterers.ClustreamClustererAdapter -l (org.apache.samoa.moa.clusterers.clustream.WithKmeans  -m 100 -k "+numClusters+")))";
+        //query ="org.wso2.carbon.ml.siddhi.extension.streaming.samoa.StreamingClusteringTask -f "+batchSize+" -i "+maxNumEvents+" -s  (org.wso2.carbon.ml.siddhi.extension.streaming.samoa.StreamingClusteringStream -K "+numClusters+" -a "+numAttributes+") -l (org.apache.samoa.learners.clusterers.simple.DistributedClusterer -l (org.apache.samoa.learners.clusterers.ClustreamClustererAdapter -l (org.apache.samoa.moa.clusterers.clustream.WithKmeans  -m 100 -k "+numClusters+")))";
+        //query = "org.apache.samoa.tasks.ClusteringEvaluation";
+        //query = "org.wso2.carbon.ml.siddhi.extension.streaming.samoa.StreamingClusteringTask";
+        query ="org.wso2.carbon.ml.siddhi.extension.streamingml.samoa.StreamingClusteringTask -f "+batchSize+" -i "+maxNumEvents+" -s  (org.wso2.carbon.ml.siddhi.extension.streamingml.samoa.StreamingClusteringStream -K "+numClusters+" -a "+numAttributes+") -l (org.apache.samoa.learners.clusterers.simple.DistributedClusterer -l (org.apache.samoa.learners.clusterers.ClustreamClustererAdapter -l (org.apache.samoa.moa.clusterers.clustream.WithKmeans  -m 100 -k "+numClusters+")))";
+        logger.info("QUERY: "+query);
+        String args[]={query};
+        this.initClusteringTask(args);
+    }
+
+    public void initClusteringTask(String[] args) {
+        logger.info("Initializing Samoa Clustering Topology");
```

Comment:
Logs that are important to developers for debugging (issues) purpose, should go to debug level. Logs that are important to users/admins, can be in info level.
However, It is OK to have one-time initializing logs, at info level.